src/main/java/com/powsybl/openloadflow/NetworkCache.java

```diff
@@ -134,8 +135,18 @@ private boolean onGeneratorUpdate(Generator generator, String attribute, Object
                 if (attribute.equals("targetV")) {
                     double valueShift = (double) newValue - (double) oldValue;
                     GeneratorVoltageControl voltageControl = lfBus.getGeneratorVoltageControl().orElseThrow();
-                    double newTargetV = voltageControl.getTargetValue() + valueShift / lfBus.getNominalV();
-                    voltageControl.setTargetValue(newTargetV);
+                    double nominalV = voltageControl.getControlledBus().getNominalV();
+                    double newTargetV = voltageControl.getTargetValue() + valueShift / nominalV;
+                    LfNetworkParameters networkParameters = context.getParameters().getNetworkParameters();
+                    if (AbstractLfGenerator.checkTargetV(generator.getId(), newTargetV, nominalV, networkParameters, null)) {
+                        voltageControl.setTargetValue(newTargetV);
+                    } else {
+                        context.getNetwork().getGeneratorById(generator.getId()).setGeneratorControlType(LfGenerator.GeneratorControlType.OFF);
+                        if (lfBus.getGenerators().stream().noneMatch(gen -> gen.getGeneratorControlType() == LfGenerator.GeneratorControlType.VOLTAGE)) {
+                            lfBus.setGeneratorVoltageControlEnabled(false);
+                        }
+                    }
+                    context.getNetwork().validate(LoadFlowModel.AC, null);
```

Comment:
Just a question: what is the behavior of the code in case of a fast restart with a targetV change in a DC loadflow mode?

Reply:
I have checked, all is good in DC mode. No cache used.